packages/vscode-extension/lib/preview.js

```diff
@@ -1,36 +1,31 @@
 const { AppRegistry, View } = require("react-native");
 
-global.__RNIDE_previews ||= new Map();
+export const PREVIEW_APP_KEY = "RNIDE_preview";
 
-function stringifyProps(obj) {
-  const keyValuePairs = [];
+global.__RNIDE_previews ||= new Map();
 
-  for (let key in obj) {
-    if (obj.hasOwnProperty(key)) {
-      keyValuePairs.push(`${key}=${obj[key]}`);
-    }
+export function Preview({ previewKey }) {
+  const previewData = global.__RNIDE_previews.get(previewKey);
+  if (!previewData || !previewData.component) {
+    return null;
   }
-
-  return keyValuePairs.join(" ");
+  return (
+    <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
+      {previewData.component}
+    </View>
+  );
 }
 
 export function preview(component) {
-  if (component._source == null) {
+  if (!component || component._source === null) {
```

Comment:
I think we may want to have `==` check here. I think `_source` could be `undefined` here too, and since it's internal field I wouldn't rely on it being set to `null`